### PR TITLE
fix 'consul reload' on custom rpc port

### PR DESCRIPTION
**Pull request by mdelagrange**
Similar to #173.

### CODE REVIEW

File: manifests/reload_service.pp [hunk 1]
```diff
@@ -13,6 +13,9 @@
     exec { 'reload consul service':
       path        => [$consul::bin_dir,'/bin','/usr/bin'],
       command     => 'consul reload',
+      environment => [
+        "CONSUL_RPC_ADDR=${consul::rpc_addr}:${consul::rpc_port}",
```
Comment (aj-jester) [inline manifests/reload_service.pp]: can we just do this as part of the command?

```
command => "consul reload -rpc-addr=${consul::rpc_addr}:${consul::rpc_port}",
```

**Comment by mdelagrange** [comment]
@ericpfisher pointed out that, if we used the environment variable rather than the flag, we could probably make the init scripts cleaner.

**Comment by aj-jester** [comment]
@mdelagrange Can we add a test for the `elsif`? https://github.com/solarkennedy/puppet-consul/blob/master/manifests/init.pp#L133

Also it would be awesome, but not required, if we can add a test for the `sysv` (your previous PR) for this `elsif` conditional as well. I'll do the same for the `upstart` :smile:.

> if we used the environment variable rather than the flag, we could probably make the init scripts cleaner.

Hmmm perhaps but I prefer if we are more explicit with the flag :confused:

**Comment by mdelagrange** [comment]
OK

**Comment by mdelagrange** [comment]
All set.

**Comment by aj-jester** [comment]
@mdelagrange this looks great :+1: 

:bullettrain_side: LGTM

**Comment by solarkennedy** [comment]
Thanks!

**Pull request merged**